ksqldb-engine/src/test/java/io/confluent/ksql/function/UdtfLoaderTest.java

```diff
@@ -227,6 +230,64 @@ public void shouldLoadVarArgsMethod() {
     assertThat(function.getReturnType(args), equalTo(STRUCT_SCHEMA));
   }
 
+  @Test
+  public void shouldNotLetBadUdtfsExit() {
+    // We do need to set up the ExtensionSecurityManager for our test.
+    // This is controlled by a feature flag and in this test, we just directly enable it.
+    SecurityManager manager = System.getSecurityManager();
+    System.setSecurityManager(ExtensionSecurityManager.INSTANCE);
+
+    final Exception e1 = assertThrows(
+        KsqlException.class,
+        () ->
+            FUNC_REG.getTableFunction(
+                    FunctionName.of("bad_test_udtf"),
+                    Collections.singletonList(SqlArgument.of(SqlTypes.decimal(2,0))))
+                .getReturnType(ImmutableList.of(SqlArgument.of(SqlTypes.DOUBLE)))
+    );
+    assertThat(e1.getMessage(), containsString(
+        "Cannot invoke the schema provider method provideSchema for UDF bad_test_udtf."));
+
+    final Exception e2 = assertThrows(
+        KsqlFunctionException.class,
+        () ->
+            FUNC_REG.getTableFunction(
+                FunctionName.of("bad_test_udtf"),
+                Collections.singletonList(SqlArgument.of(SqlTypes.STRING))).apply("foo")
+    );
+    assertThat(e2.getMessage(), containsString(
+        "Failed to invoke function public java.util.List "
+            + "io.confluent.ksql.function.udf.BadTestUdtf.listStringReturn(java.lang.String)"));
+
+    // Stop reflection
+    /*
+    final Exception e3 = assertThrows(
+        KsqlFunctionException.class,
+        () ->
+            FUNC_REG.getTableFunction(
+                FunctionName.of("bad_test_udtf"),
+                Collections.singletonList(SqlArgument.of(SqlTypes.BOOLEAN))).apply(true)
+    );
+    assertThat(e3.getMessage(), containsString(
+        "Failed to invoke function public java.util.List "
+            + "io.confluent.ksql.function.udf.BadTestUdtf.listBooleanReturn(boolean)"));
+    */
```

Comment:
This test case shows that the ExtensionSecurityManager needs to handle reflection.  It is commented out since that case still needs to be addressed.  

I can remove it.